### PR TITLE
1452 Refactor module resolver

### LINKED ISSUE #1452: Refactor paths to resolve for ui and global components
As part of *issue for refactoring paths* and future refactoring, we'll be moving files around. We're currently using relative paths for client-side components, which is a pain to update when refactoring.

We should use Webpack's resolver to create top level paths, so we don't have to rely on relative paths to commonly used components, such as `ui/` and `global/`

See [here](https://github.com/elifesciences/elife-xpub/blob/develop/app/components/pages/SubmissionWizard/steps/Editors/EditorsPage.js#L8-L12) for an example of some relative pathing we want to avoid.

### PR DESCRIPTION
**Pull request by SemraSh**
#### Background

The first step of refactoring, using ui and global as modules within the app.

#### Any relevant tickets

Closes #1452

### CODE REVIEW

**Review by diversemix — APPROVED** [review]
Looks good to me

**Comment by diversemix** [comment]
Let's merge in the morning... as this is the first after release @hdrury

File: .eslintrc [hunk 1]
```diff
@@ -3,34 +3,49 @@
...
   "parser": "babel-eslint",
   "rules": {
     "global-require": 0,
-    "class-methods-use-this": ["error", {
-       "exceptMethods": ["storage",
-        "render",
-        "componentDidMount",
-        "shouldComponentUpdate"]
-     }],
+    "import/no-unresolved": "off",
```
Comment (Cooryd) [inline .eslintrc:17]: Is there any way to whitelist this rule? I feel we'll be losing some useful checks by disabling it globally just to allow aliases for the ui components. Maybe we can look into creating `package.json` files for the ui molecules similar to how we're using the `xpub-server` and `xpub-model` in the server?

Comment (diversemix) [reply .eslintrc:17]: Fair point ... I did scratch my head over this when reviewing ... glad you picked it up @Cooryd

Comment (hem-brahmbhatt) [reply .eslintrc:17]: I think this actually can be whitelisted, it was initially put in as an attempt to achieve the changes we wanted but then we found another way by specifying the actual resolver.

Comment (SemraSh) [reply .eslintrc:17]: Creating a pr to fix this.

**Pull request merged**